sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharpExplodedGraph.cs

```diff
@@ -549,8 +549,6 @@ private ProgramState EnsureStackState(ExpressionSyntax parenthesizedExpression,
             if (ShouldConsumeValue(parenthesizedExpression))
             {
                 var newProgramState = programState.PopValue();
-                System.Diagnostics.Debug.Assert(!newProgramState.HasValue);

```

Comment:
why did you remove the assert? it doesn't affect prod, it will only run if compiling with `/d:DEBUG`